python/tests/test_account.py

```diff
@@ -1970,6 +1970,42 @@ def test_set_get_group_image(self, acfactory, data, lp):
         assert msg_back.chat == chat
         assert chat.get_profile_image() is None
 
+    def test_fetch_deleted_msg(self, acfactory, lp):
+        ac1 = acfactory.get_one_online_account()
+        ac1.stop_io()
+
+        ac1.direct_imap.append("INBOX", """
+            From: alice <alice@example.org>
+            Subject: subj
+            To: bob@example.com
+            Chat-Version: 1.0
+            Message-ID: <aepiors@example.org>
+            Content-Type: text/plain; charset=utf-8
+
+            Deleted message
+        """)
+        ac1.direct_imap.delete("1:*", expunge=False)
+        ac1.start_io()
+
+        for ev in ac1._evtracker.iter_events():
+            if ev.name == "DC_EVENT_MSGS_CHANGED":
+                pytest.fail("A deleted message was shown to the user")
+
+            if ev.name == "DC_EVENT_INFO" and "1 mails read from" in ev.data2:
```

Comment:
Better check with `ev.data2.startswith("1 mails read from")`, to make sure it's not "11 mails read from" or something like this.

We have low-level protocol debugging events like `DC_EVENT_IMAP_MESSAGE_MOVED`, `DC_EVENT_SMTP_MESSAGE_SENT` etc., which are not used in the UI. I think this test justifies adding another event (`DC_EVENT_IMAP_FETCH` or something similar) with the UID of fetched message as data1 and folder name as data2, so you don't need to "parse" log strings. On the other hand any change in INFO strings will be caught by tests, so up to you if you want to add it.

Reply:
The problem is, the whole string is `src/imap.rs:734: 1 mails read from "INBOX".`, so it doesn't start with this.

What I could do is sth. like `" 1 mails read from" in ev.data2` (add a space before the 1). This will look a bit weird but work. OTOH, where should 11 mails come from.